zio-schema/shared/src/main/scala/zio/schema/Ast.scala

```diff
@@ -0,0 +1,293 @@
+package zio.schema
+
+import scala.collection.immutable.ListMap
+
+import zio.{ Chunk, ChunkBuilder }
+
+sealed trait Ast { self =>
```

Comment:
My main concern here is with the name `Ast`. Because we are in the `zio.schema` package, it would be nice if we can shoot for `import zioi.schema._` in most code bases. But the name `Ast` is quite common.

Reply:
Yeah, that's a good paint. Maybe `SchemaAst`?